wrappers/src/scheduler_cs.cpp

```diff
@@ -100,22 +96,31 @@ struct ThreadScheduler : public Scheduler {
 
 extern "C" {
 
-REALM_EXPORT void realm_install_scheduler_callbacks(GetContextT* get, PostOnContextT* post, ReleaseContextT* release, IsOnContextT* is_on_context)
+REALM_EXPORT void realm_install_scheduler_callbacks(GetContextT* get, PostOnContextT* post, ReleaseContextT* release, IsOnContextT* is_on)
 {
-    Scheduler::set_default_factory([=]() -> std::unique_ptr<Scheduler> {
-        void* context = get();
+    s_get_context = wrap_managed_callback(get);
+    s_post_on_context = wrap_managed_callback(post);
+    s_release_context = wrap_managed_callback(release);
+    s_is_on_context = wrap_managed_callback(is_on);
+
+    Scheduler::set_default_factory([]() -> std::unique_ptr<Scheduler> {
+        void* context = s_get_context();
         if (context) {
-            return std::make_unique<SynchronizationContextScheduler>(context, post, release, is_on_context);
+            return std::make_unique<SynchronizationContextScheduler>(context);
         }
 
         return std::make_unique<ThreadScheduler>();
     });
 }
 
-REALM_EXPORT void realm_scheduler_invoke_function(void* function_ptr)
+REALM_EXPORT void realm_scheduler_invoke_function(void* function_ptr, bool execute_func)
```

Comment:
`execute_func` will be false if the context was invalidated before invoke_function was called. In this case, we simply delete the allocated function.